drivers/clock_control/clock_control_tt_bh.c

```diff
@@ -475,6 +475,11 @@ static int clock_control_tt_bh_set_rate(const struct device *dev, clock_control_
 			clock_control_tt_bh_write_reg(config, PLL_CNTL_1_OFFSET, pll_cntl_1.val);
 			k_busy_wait_ns(100);
 		}
+	} else if (clock == CLOCK_CONTROL_TT_BH_INIT_STATE) {
```

Comment:
This seems like a good use case for deferred init: https://docs.zephyrproject.org/latest/kernel/drivers/index.html#deferred-initialization

If we used this here, could we just defer the init until we need to put the PLL into initial state? Or do we need to do the device init separately from setting the initial PLL settings?